src/linker/Linker.Dataflow/DynamicallyAccessedMembersBinder.cs

```diff
@@ -327,5 +334,56 @@ public static IEnumerable<InterfaceImplementation> GetAllInterfaceImplementation
 				type = context.TryResolve (type.BaseType);
 			}
 		}
+
+		public static IEnumerable<IMetadataTokenProvider> GetAllOnType (this TypeDefinition type, LinkContext context) => GetAllOnType (type, context, new HashSet<TypeDefinition> ());
+
+		static IEnumerable<IMetadataTokenProvider> GetAllOnType (TypeDefinition type, LinkContext context, HashSet<TypeDefinition> types)
+		{
+			if (!types.Add (type))
+				yield break;
+
+			if (type.HasNestedTypes) {
+				foreach (var nested in type.NestedTypes) {
+					yield return nested;
+					foreach (var m in GetAllOnType (nested, context, types))
+						yield return m;
```

Comment:
Nit: it might be better to just have an extra `List<IMetadataTokenProvider>` parameter that we add do instead of using `yield return`.

Nested/recursive yield returns have a funny debug experience.